packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3402,9 +3412,14 @@ export class ContainerRuntime
 	 * all references added in the system.
 	 * @param fromPath - The absolute path of the node that added the reference.
 	 * @param toPath - The absolute path of the outbound node that is referenced.
+	 * @param messageTimestampMs - The timestamp of the message that added the reference.
 	 */
-	public addedGCOutboundRoute(fromPath: string, toPath: string) {
-		this.garbageCollector.addedOutboundReference(fromPath, toPath);
+	public addedGCOutboundRoute(fromPath: string, toPath: string, messageTimestampMs?: number) {
+		this.garbageCollector.addedOutboundReference(
+			fromPath,
+			toPath,
+			messageTimestampMs ?? this.getCurrentReferenceTimestampMs(),
```

Comment:
So I'm tracing around to find when `messageTimestampeMs` is missing, I found only one spot which is when trying to set an alias when detached.  I think it would be better to just not call `addedGCOutboundRoute` if the timestamp is missing in that case (put a comment saying it means we're detached?), and then everywhere else can require the timestamp.

Hopefully I didn't miss anything with that, take a look.

Reply:
The only other case is back-compat across data store runtime -> data store context. Since I added timestampMs to `addedGCOutboundRoute` in this PR, old data store runtimes won't send it.

Reply:
I will create a task to make this required once the back-compat concern goes away.

Reply:
I added an assert here that timestamp is required. Looks safe to me, can you please double check.

Reply:
AB#8329

Reply:
The assert can be removed once `messageTimestampMs` is required on `IFluidParentContext`

Reply:
Removed the assert and added a telemetry log instead since it's not worth the risk. The log will tell us if this case is hit and we can remove it once we move past the back-compat concerns.